### ISSUE TITLE
black output with bf16

### ISSUE THREAD

**Issue by foxender** [kijai/ComfyUI-HunyuanVideoWrapper#105]
hi, i use bf16 still has a black output(torch2.5.1+cuda12.4)
Failed to validate prompt for output 34:
* HyVideoVAELoader 7:
  - Value not in list: model_name: 'hyvid\hunyuan_video_vae_bf16.safetensors' not in ['ae.sft', 'animevae.pt', 'diffusion_pytorch_model.safetensors', 'greyVAEForSDXL_t.pt', 'hunyuan_video_vae_bf16.safetensors', 'hunyuan_video_vae_fp32.safetensors', 'madebyollin-sdxl-vae-fp16-fix.safetensors', 'mochi\\mochi_preview_vae_decoder_bf16.safetensors', 'mochi\\mochi_vae.safetensors', 'orangemix.vae.pt', 'pyramid_flow_vae_bf16.safetensors', 'sdxlVAE_sdxlVAE.safetensors', 'vae-ft-mse-840000-ema-pruned.ckpt', 'vae-ft-mse-840000-ema-pruned.safetensors', 'xlVAEC_e7.safetensors', 'xlVAEC_f1.safetensors']
* HyVideoModelLoader 1:
  - Value not in list: model: 'hyvideo\hunyuan_video_720_fp8_e4m3fn.safetensors' not in (list of length 22)
Output will be ignored
invalid prompt: {'type': 'prompt_outputs_failed_validation', 'message': 'Prompt outputs failed validation', 'details': '', 'extra_info': {}}
got prompt
Loading text encoder model (clipL) from: E:\aigc\ComfyUI_windows_portable\ComfyUI\models\clip\clip-vit-large-patch14
Text encoder to dtype: torch.float16
Loading tokenizer (clipL) from: E:\aigc\ComfyUI_windows_portable\ComfyUI\models\clip\clip-vit-large-patch14
Loading text encoder model (llm) from: E:\aigc\ComfyUI_windows_portable\ComfyUI\models\LLM\llava-llama-3-8b-text-encoder-tokenizer
Loading checkpoint shards: 100%|██████████| 4/4 [00:13<00:00,  3.31s/it]
Text encoder to dtype: torch.float16
Loading tokenizer (llm) from: E:\aigc\ComfyUI_windows_portable\ComfyUI\models\LLM\llava-llama-3-8b-text-encoder-tokenizer
llm prompt attention_mask shape: torch.Size([1, 161]), masked tokens: 31
clipL prompt attention_mask shape: torch.Size([1, 77]), masked tokens: 32
Using accelerate to load and assign model weights to device...
Scheduler config: FrozenDict([('num_train_timesteps', 1000), ('shift', 9.0), ('reverse', True), ('solver', 'euler'), ('n_tokens', None), ('_use_default_values', ['num_train_timesteps', 'n_tokens'])])
Input (height, width, video_length) = (320, 512, 85)
Sampling 85 frames in 22 latents at 512x320 with 30 inference steps
100%|██████████| 30/30 [04:55<00:00,  9.84s/it]
Allocated memory: memory=12.931 GB
Max allocated memory: max_memory=15.402 GB
Max reserved memory: max_reserved=16.625 GB
Decoding rows: 100%|██████████| 2/2 [02:12<00:00, 66.08s/it]
Blending tiles: 100%|██████████| 2/2 [00:00<00:00, 69.15it/s]
Decoding rows: 100%|██████████| 2/2 [00:14<00:00,  7.28s/it]
Blending tiles: 100%|██████████| 2/2 [00:02<00:00,  1.42s/it]
Prompt executed in 492.81 seconds
![image](https://github.com/user-attachments/assets/eadae288-bc12-46b3-8643-aa8eae16b0e3)

**Comment by AIGC101** [comment]
same here

**Comment by foxender** [comment]
The problem has been solved, not just the model, all parts related to precision (including TextENCODER) need to be set to bf16

**Comment by foxender** [comment]
> 这里也是一样的

solved
